apps/prairielearn/src/lib/socket-server.ts

```diff
@@ -77,5 +77,6 @@ export async function close() {
   await Promise.all([...io._nsps.values()].map((nsp) => nsp.adapter.close()));
   io.engine.close();
 
-  await Promise.all([pub?.quit(), sub?.quit()]);
+  await pub?.quit();
+  await sub?.quit();
```

Comment:
I think this might sorta-kinda fix things by letting the `sub` queue finish processing unsubscribes before actually closing down the queues.

Alternatively, I bet that `await`ing the unsubscribes here would help, but that's outside our immediate control.

https://github.com/socketio/socket.io-redis-adapter/blob/cdb55353f83c78cabe9788683e4dd93ac4cd50c9/lib/index.ts#L897-L944